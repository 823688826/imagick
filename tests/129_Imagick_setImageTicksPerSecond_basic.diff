001+ Fatal error: Uncaught exception 'ImagickException' with message 'Can not process empty Imagick object' in /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php:10
001- Ok
002+ Stack trace:
003+ #0 /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php(10): Imagick->valid()
004+ #1 /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php(29): setImageTicksPerSecond()
005+ #2 {main}
006+   thrown in /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php on line 10