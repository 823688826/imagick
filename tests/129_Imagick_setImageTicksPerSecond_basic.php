<?php


function setImageTicksPerSecond() {

    $imagick = new \Imagick(realpath("images/coolGif.gif"));
    $totalFrames = $imagick->getNumberImages();
    $frameCount = 0;

    foreach ($imagick as $frame) {
        /** @var $frame \Imagick */

        if ($frameCount < ($totalFrames / 2)) {
            //Modify the frame to be displayed for twice as long as it currently is.
            $frame->setImageTicksPerSecond(50);
        }
        else {
            //Modify the frame to be displayed for half as long as it currently is.
            $frame->setImageTicksPerSecond(200);
        }
        $frameCount++;
    }

    $imagick2 = $imagick->deconstructImages();
    $bytes = $imagick2->getImagesBlob();
    if (strlen($bytes) <= 0) { echo "Failed to generate image.";} 
}

setImageTicksPerSecond() ;
echo "Ok";
?>
