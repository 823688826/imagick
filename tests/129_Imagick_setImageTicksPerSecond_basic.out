Fatal error: Uncaught exception 'ImagickException' with message 'Can not process empty Imagick object' in /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php:10
Stack trace:
#0 /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php(10): Imagick->valid()
#1 /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php(29): setImageTicksPerSecond()
#2 {main}
  thrown in /home/github/mkoppanen-imagick/tests/129_Imagick_setImageTicksPerSecond_basic.php on line 10